Integrations/integration-PhishMe.yml

```diff
@@ -107,25 +158,33 @@ script:
         var md = "## PhishMe URL Reputation for: "+url+"\n";
         var dbotScore = 0;
         var threatArray = [];
-        if (threats.length !== 0) {
+        var threshold = params.urlThreshold;
+        var indicatorFnd = false;
+        if (threats.length && threats.length !== 0) {
             ec.URL = {
                         Data: url,
                         Malicious: {Vendor: 'PhishMe', Description: 'Match found in PhishMe database'}
                       };
             for (var k = 0; k < threats.length; k++) {
-                for (var i = 0; i < threats[0].blockSet.length; i++){
-                    if (threats[0].blockSet[i]['data'].indexOf(url) !== -1) {
-                        var severityLevel = threats[0].blockSet[i]['impact'];
-                        break;
-                    }
-                }
-                dbotScore = calcScore(severityLevel);
-                md += 'Verdict: ' + calcVerdict(dbotScore) +'\n';
-                md += addMD(threats, k);
+                threatAnalysis = checkThreat(threats[k], threshold, url);
                 threatArray[k] = threats[k].id;
+                md += threatAnalysis['md'];
+                if (threatAnalysis.indicatorFound) {
+                    indicatorFnd = true;
+                    dbotScore = threatAnalysis['severityScore'];
+                } else if (!indicatorFnd) {
+                    dbotScore = Math.max(dbotScore, threatAnalysis['severityScore']);
+                }
             }
             ec.URL['PhishMe'] = {};
             ec.URL.PhishMe['ThreatIDs'] = threatArray;
+            ec.PhishMe = {
+                URL: {
+                    Data: url,
+                    Malicious: ec.URL.Malicious,
```

Comment:
not sure if Malicious key should be added anyway.

Reply:
You're right, I'll add in the logic